hudi-utilities/src/test/java/org/apache/hudi/utilities/inline/fs/TestHFileReadWriteFlow.java

```diff
@@ -0,0 +1,243 @@
+        .withFileContext(meta)
+        .withComparator(new KeyValue.KVComparator())
+        .create();
+
+    writeRecords(writer);
+    fout.close();
+
+    byte[] inlineBytes = getBytesToInline(outerInMemFSPath);
+    long startOffset = generateOuterFile(outerPath, inlineBytes);
+
+    long inlineLength = inlineBytes.length;
+
+    // Generate phantom inline file
+    Path inlinePath = getPhantomFile(outerPath, startOffset, inlineLength);
+
+    InlineFileSystem inlineFileSystem = (InlineFileSystem) inlinePath.getFileSystem(inlineConf);
+    FSDataInputStream fin = inlineFileSystem.open(inlinePath);
+
+    HFile.Reader reader = HFile.createReader(inlineFileSystem, inlinePath, cacheConf, inlineConf);
+    // Load up the index.
+    reader.loadFileInfo();
+    // Get a scanner that caches and that does not use pread.
+    HFileScanner scanner = reader.getScanner(true, false);
+    // Align scanner at start of the file.
+    scanner.seekTo();
+    readAllRecords(scanner);
+
+    Set<Integer> rowIdsToSearch = getRandomValidRowIds(10);
+    for (int rowId : rowIdsToSearch) {
+      Assert.assertTrue("location lookup failed",
+          scanner.seekTo(KeyValue.createKeyValueFromKey(getSomeKey(rowId))) == 0);
+      // read the key and see if it matches
+      ByteBuffer readKey = scanner.getKey();
+      Assert.assertTrue("seeked key does not match", Arrays.equals(getSomeKey(rowId),
+          Bytes.toBytes(readKey)));
+      scanner.seekTo(KeyValue.createKeyValueFromKey(getSomeKey(rowId)));
+      ByteBuffer val1 = scanner.getValue();
+      scanner.seekTo(KeyValue.createKeyValueFromKey(getSomeKey(rowId)));
+      ByteBuffer val2 = scanner.getValue();
+      Assert.assertTrue(Arrays.equals(Bytes.toBytes(val1), Bytes.toBytes(val2)));
+    }
+
+    int[] invalidRowIds = {-4, maxRows, maxRows + 1, maxRows + 120, maxRows + 160, maxRows + 1000};
+    for (int rowId : invalidRowIds) {
+      Assert.assertFalse("location lookup should have failed",
+          scanner.seekTo(KeyValue.createKeyValueFromKey(getSomeKey(rowId))) == 0);
+    }
+    reader.close();
+    fin.close();
+    outerPath.getFileSystem(inMemoryConf).delete(outerPath, true);
+  }
+
+  private Set<Integer> getRandomValidRowIds(int count) {
+    Set<Integer> rowIds = new HashSet<>();
+    while (rowIds.size() < count) {
+      int index = RANDOM.nextInt(maxRows);
+      if (!rowIds.contains(index)) {
+        rowIds.add(index);
+      }
+    }
+    return rowIds;
+  }
+
+  private byte[] getSomeKey(int rowId) {
+    KeyValue kv = new KeyValue(String.format(LOCAL_FORMATTER, Integer.valueOf(rowId)).getBytes(),
+        Bytes.toBytes("family"), Bytes.toBytes("qual"), HConstants.LATEST_TIMESTAMP, KeyValue.Type.Put);
+    return kv.getKey();
+  }
+
+  private FSDataOutputStream createFSOutput(Path name, Configuration conf) throws IOException {
+    //if (fs.exists(name)) fs.delete(name, true);
+    FSDataOutputStream fout = name.getFileSystem(conf).create(name);
+    return fout;
```

Comment:
single line return?

Reply:
Sorry had some debug statements before and hence. will fix it.